pkg/networking/middleware/retry_middleware.go

```diff
@@ -22,15 +22,22 @@ const ConfigurationKeyRetryAttempts = "internal_network_request_max_attempts"
 const configurationKeyRetryAfter = "internal_network_request_retry_after_seconds"
 const retryCountHeaderKey = "Snyk-Request-Attempt-Count"
 
-// This lookup table defines the response status codes that should be retried
-var statusCodesToRetryLUT = map[int]bool{
-	http.StatusTooManyRequests:     true,
-	http.StatusTooEarly:            true,
-	http.StatusRequestTimeout:      true,
-	http.StatusInternalServerError: true,
-	http.StatusBadGateway:          true,
-	http.StatusServiceUnavailable:  true,
-	http.StatusGatewayTimeout:      true,
+type RetryLogic struct {
+	shouldRetry      bool
+	maxRetryOverride *uint
+}
+
+var attemptTwoNetworkRequests uint = 2
```

Comment:
Suggestion: I'm wondering if we should spend one more attempt here just to be a bit safer.